### PR TITLE
test: Add unit tests to new integration directory components

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
Add unit tests for:
- integrationListDirectory component
- IntegrationRow component

## Test Plan
Acceptance Criteria: 60% unit test coverage for the two components above

### CODE REVIEW

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        routerContext
+      );
+
+      expect(wrapper.find('SearchInput').exists()).toBeTruthy();
+      expect(wrapper.find('PanelBody').exists()).toBeTruthy();
+      expect(wrapper.find('IntegrationRow')).toHaveLength(5);
+      expect(
+        wrapper
+          .find('IntegrationRow')
+          .first()
```
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: @NisanthanNanthakumar It would be good to check the order of all the integration slugs

**Comment by scefali** [comment]
@NisanthanNanthakumar You should add a snapshot test for both files

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        [
+          `/organizations/${org.slug}/sentry-apps/`,
+          [
+            {
+              allowedOrigins: [],
+              author: 'Sentry',
+              clientId:
+                'a6d35972d4164ef18845b1e2ca954fe70ac196e0b20d4d1e8760a38772cf6f1c',
+              clientSecret:
+                '8f47dcef40f7486f9bacfeca257022e092a483add7cf4d619993b9ace9775a79',
```
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: @NisanthanNanthakumar is this your actual client secret lol? Let's not expose that in a test...

File: tests/js/spec/views/organizationIntegrations/integrationRow.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      );
+      expect(wrapper.find('IntegrationStatus').props().status).toEqual('Installed');
+      expect(wrapper.find('PublishStatus').exists()).toEqual(false);
+      expect(wrapper.find('StyledLink').props().children).toEqual('1 Configuration');
+    });
+
+    it('has not been installed', async function() {
+      const wrapper = mountWithTheme(
+        <IntegrationRow
+          organization={org}
```
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/integrationRow.spec.jsx]: @NisanthanNanthakumar We should have a test with >1 configurations. Also, we should check the configuration count that's visible matches the prop value.

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx (hunk 2 above)
Comment (NisanthanNanthakumar) [reply tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: @scefali it's a client secret for a fake internal app on my local instance

Comment (scefali) [reply tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: Fair enough :)

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+      expect(wrapper.find('SearchInput').exists()).toBeTruthy();
+      expect(wrapper.find('PanelBody').exists()).toBeTruthy();
+      expect(wrapper.find('IntegrationRow')).toHaveLength(5);
+      expect(
+        wrapper
+          .find('IntegrationRow')
+          .first()
+          .props().slug
+      ).toEqual('bitbucket');
```
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: @NisanthanNanthakumar We should also have a test for the legacy plugin functionality. So the Github plugin should only show up if it's installed and it should have `(Legacy)` in the name.

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+          .find('IntegrationRow')
+          .at(3)
+          .props().slug
+      ).toEqual('amazon-sqs');
+      expect(
+        wrapper
+          .find('IntegrationRow')
+          .at(4)
+          .props().slug
+      ).toEqual('clickup');
```
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: @NisanthanNanthakumar Nit: there is probably a more concise way to do these assertions with a for loop over an array of the slugs you expect in order.

**Review by scefali — APPROVED** [review]
LGTM, just one optional suggestion about refactoring a test

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+  let wrapper;
+
+  describe('Renders view', function() {
+    beforeEach(() => {
+      mockResponse([
+        [
+          `/organizations/${org.slug}/config/integrations/`,
+          {
+            providers: [
+              {
```
Comment (billyvg) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: @NisanthanNanthakumar can we move these to `tests/js/fixtures` - I think we have some there for github, it would make these tests a bit easier to read

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+    });
+  });
+};
+
+describe('IntegrationListDirectory', function() {
+  beforeEach(function() {
+    Client.clearMockResponses();
+  });
+
+  const org = TestStubs.Organization();
```
Comment (billyvg) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: I would look at `sentry-test/initializeOrg` - it sets up org in context (even though it's not needed with these tests)

File: tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+        [`/organizations/${org.slug}/repos/?status=unmigratable`, []],
+      ]);
+
+      wrapper = mountWithTheme(
+        <IntegrationListDirectory params={{orgId: org.slug}} location={{search: ''}} />,
+        routerContext
+      );
+    });
+    it('match previous snapshot', async function() {
+      expect(wrapper).toMatchSnapshot();
```
Comment (billyvg) [inline tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx]: I'm skeptical of how useful these snapshots are, imo it's more painful than useful, especially when you're snapshotting a large component with a 1k+ line snapshot file. Your test below tells me a lot more about what you deem important enough in the view that you're writing an explicit expectation for, whereas this snapshot can throw lots of false negatives (e.g. emotion class name changes, data changes).

File: tests/js/spec/views/organizationIntegrations/integrationRow.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+      );
+      expect(wrapper.find('PluginIcon').props().pluginId).toEqual('bitbucket');
+      expect(wrapper.find('IntegrationName').props().children).toEqual('Bitbucket');
+      expect(wrapper.find('IntegrationName').props().to).toEqual(
+        `/settings/${org.slug}/integrations/bitbucket/`
+      );
+      expect(wrapper.find('IntegrationStatus').props().status).toEqual('Installed');
+      expect(wrapper.find('PublishStatus').exists()).toEqual(false);
+      expect(wrapper.find('StyledLink').props().children).toEqual('1 Configuration');
+      expect(wrapper).toMatchSnapshot();
```
Comment (billyvg) [inline tests/js/spec/views/organizationIntegrations/integrationRow.spec.jsx]: Why snapshot this and not the SentryApp tests?  The snapshots seems a bit redundant given the above expectations.

File: tests/js/spec/views/organizationIntegrations/integrationRow.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+          configurations={0}
+          data-test-id="integration-row"
+        />,
+        routerContext
+      );
+      expect(wrapper.find('PluginIcon').props().pluginId).toEqual('amazon-sqs');
+      expect(wrapper.find('IntegrationName').props().children).toEqual('Amazon SQS');
+      expect(wrapper.find('IntegrationName').props().to).toEqual(
+        `/settings/${org.slug}/plugins/amazon-sqs/`
+      );
```
Comment (billyvg) [inline tests/js/spec/views/organizationIntegrations/integrationRow.spec.jsx]: These seem a bit irrelevant to what you're testing (and it's been tested throughout)

**Review by billyvg — COMMENTED** [review]
@NisanthanNanthakumar Left some notes -- I would say generally these types of tests where you're testing visual structure is better suited for percy snapshots, but I'm guessing it's a bit of a pita to mock the integration endpoints in our acceptance tests, so having these tests here makes sense.
